### PR TITLE
CIV-6352 BundleIntegrationMain

### PR DESCRIPTION
**Pull request by bhagyashreesharma90**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-6352

### Change description ###
Bundle Integration consolidated changes for all tickets

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/helpers/BundleRequestMapper.java [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+                                                                 .documentUrl(document.getValue().getDocumentUrl())
+                                                                 .documentBinaryUrl(document.getValue().getDocumentBinaryUrl())
+                                                                 .documentFilename(document.getValue().getDocumentFileName()).build())
+                                               .build());
+
+            });
+        }
+        List<Element<BundlingRequestDocument>> particulars = ElementUtils.wrapElements(bundlingServedDocFiles);
+        return  ServedDocument.builder().particularsOfClaimDocument(particulars).build();
+    }
```
Comment (filippo-possenti-version1) [inline src/main/java/uk/gov/hmcts/reform/civil/helpers/BundleRequestMapper.java:176]: Try short-circuiting the "if" statement in this function.

With that done, you'll notice that ultimately you just want to process things if they exist or work against an empty collection otherwise.

It may lead you to the following:
```
    private ServedDocument mapServedDocuments(ServedDocumentFiles servedDocumentFiles) {
        List<BundlingRequestDocument> bundlingServedDocFiles = Optional.of(servedDocumentFiles)
            .map(ServedDocumentFiles::getParticularsOfClaimDocument)
            .orElse(new ArrayList<>())
            .stream()
            .map(document -> BundlingRequestDocument.builder()
                .documentFileName(document.getValue().getDocumentFileName())
                .documentLink(DocumentLink.builder()
                                  .documentUrl(document.getValue().getDocumentUrl())
                                  .documentBinaryUrl(document.getValue().getDocumentBinaryUrl())
                                  .documentFilename(document.getValue().getDocumentFileName()).build())
                .build()).collect(Collectors.toList());

        List<Element<BundlingRequestDocument>> particulars = ElementUtils.wrapElements(bundlingServedDocFiles);
        return ServedDocument.builder().particularsOfClaimDocument(particulars).build();
    }
```

Whether one sees any benefit in it or not is subjective and streams can get complicated. Still, short-circuiting the "if" may improve readability.

Also consider that you're not fully covering all the branches of the "if" so I will want a test about the missing branch.

File: src/main/java/uk/gov/hmcts/reform/civil/helpers/BundleRequestMapper.java [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+
+    private List<Element<BundlingRequestDocument>> mapSystemGeneratedcaseDocument(List<Element<CaseDocument>> systemGeneratedCaseDocuments, Document orderSDODocumentDJ) {
+        List<BundlingRequestDocument> bundlingSystemGeneratedCaseDocs = new ArrayList<>();
+        if (!Optional.ofNullable(systemGeneratedCaseDocuments).isEmpty()) {
```
Comment (filippo-possenti-version1) [inline src/main/java/uk/gov/hmcts/reform/civil/helpers/BundleRequestMapper.java:180]: Short-circuit this "if" statement.

Before you do however, add a test around the missing branch (that is: where `systemGeneratedCaseDocuments` is empty).

File: src/main/java/uk/gov/hmcts/reform/civil/model/Bundle.java [hunk 3]
```diff
@@ -114,6 +127,18 @@ public String getFilename() {
         return filename;
     }
 
+    public Optional<LocalDateTime> getCreatedOn() {
+        return checkIsOptional(createdOn);
+    }
+
+    public Optional<LocalDate> getBundleHearingDate() {
+        return bundleHearingDate;
```
Comment (filippo-possenti-version1) [inline src/main/java/uk/gov/hmcts/reform/civil/model/Bundle.java:135]: Presumably this also will need `checkIsOptional` if its reason is that jackson may initialize `Optional` fields to null.

File: src/main/java/uk/gov/hmcts/reform/civil/model/Bundle.java [hunk 4]
```diff
@@ -114,6 +127,18 @@ public String getFilename() {
...
+    }
+
+    public Optional<String> getStitchingFailureMessage() {
+        return stitchingFailureMessage;
```
Comment (filippo-possenti-version1) [inline src/main/java/uk/gov/hmcts/reform/civil/model/Bundle.java:139]: Presumably this also will need `checkIsOptional` if its reason is that jackson may initialize `Optional` fields to null.

**Review by filippo-possenti-version1 — CHANGES_REQUESTED** [review]
Mostly I want additional test coverage and short-circuiting of the "if" statements for improved readability.
Aside from that, I have no objections.

Start with the missing test coverage. Once that is done and everything is appropriately covered, you can start short-circuiting the "if" statements.

For the branch coverage, you will need to configure your IntelliJ. Follow this guide:
https://improveandrepeat.com/2021/02/how-to-activate-branch-coverage-in-intellij/

**Review by filippo-possenti-version1 — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/handler/event/BundleCreationTriggerEventHandler.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    }
+
+    IdValue<Bundle> prepareNewBundle(uk.gov.hmcts.reform.civil.model.bundle.Bundle bundle, CaseData caseData) {
+        Bundle result = Bundle.builder()
+            .bundleHearingDate(Optional.of(caseData.getHearingDate()))
+            .stitchedDocument(Optional.ofNullable(bundle.getValue().getStitchedDocument()))
+            .filename(bundle.getValue().getFileName())
+            .title(bundle.getValue().getTitle())
+            .description(null != bundle.getValue().getDescription()
+                             ? Optional.of(bundle.getValue().getDescription()).get() : "")
```
[the remark this answers is not in the document]
Comment (filippo-possenti-version1) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/event/BundleCreationTriggerEventHandler.java:86]: Marking as resolved as at this stage I don't necessarily want to change the code.

File: src/main/java/uk/gov/hmcts/reform/civil/helpers/BundleRequestMapper.java [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+                .courtLocation(caseData.getHearingLocation().getValue().getLabel())
+                .hearingDate(null != caseData.getHearingDate()
+                                 ? DateFormatHelper.formatLocalDate(caseData.getHearingDate(), "dd-MM-yyyy") : null)
+                .ccdCaseReference(caseData.getCcdCaseReference())
+                .build();
+        bundlingCaseData = mapRespondent2Applicant2Details(bundlingCaseData, caseData);
+        return  bundlingCaseData;
+    }
+
+    private BundlingCaseData mapRespondent2Applicant2Details(BundlingCaseData bundlingCaseData, CaseData caseData) {
```
[the remark this answers is not in the document]
Comment (filippo-possenti-version1) [reply src/main/java/uk/gov/hmcts/reform/civil/helpers/BundleRequestMapper.java:137]: Marking as resolved as at this stage I don't necessarily want to change the code.

File: src/main/java/uk/gov/hmcts/reform/civil/helpers/BundleRequestMapper.java [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+        if (null != caseData.getAddApplicant2() && caseData.getAddApplicant2().equals(YesOrNo.YES)) {
+            bundlingCaseData.toBuilder().hasApplicant2(true);
+        }
+        if (null != caseData.getAddRespondent2() && caseData.getAddRespondent2().equals(YesOrNo.YES)) {
```
[the remark this answers is not in the document]
Comment (filippo-possenti-version1) [reply src/main/java/uk/gov/hmcts/reform/civil/helpers/BundleRequestMapper.java:141]: Marking as resolved as at this stage I don't necessarily want to change the code.

File: src/main/java/uk/gov/hmcts/reform/civil/helpers/BundleRequestMapper.java (hunk 2 above)
Comment (filippo-possenti-version1) [reply src/main/java/uk/gov/hmcts/reform/civil/helpers/BundleRequestMapper.java:174]: Marking as resolved as at this stage I don't necessarily want to change the code.

**Review by sankhajuria — APPROVED** [review]
